src/hifz/dataserver.py

```diff
@@ -18,8 +19,12 @@ def read_entries(self, file_path: str) -> list[Card]:
 
         try:
             with Path(file_path).open() as f:
-                entries = csv.DictReader(f)
-                cards = [Card(e["front"], e["back"]) for e in entries]
+                if file_path.lower.endswith(".csv"):
+                    entries = csv.DictReader(f)
+                    cards = [Card(e["front"], e["back"]) for e in entries]
+                elif file_path.endswith(".json"):
+                    entries = json.load(f)
+                    cards = [Card(e["front"], e["back"]) for e in entries]
```

Comment:
Alternatively, you can move cards = [Card(e["front"], e["back]) for e in entries] outside of the if-else logic.

```suggestion
                if file_path.lower.endswith(".csv"):
                    entries = csv.DictReader(f)
                elif file_path.endswith(".json"):
                    entries = json.load(f)
                cards = [Card(e["front"], e["back"]) for e in entries]
```